File: Leetcode/238-Product-of-Array-Except-Self/238-Product-of-Array-Except-Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1]
        suffix = [1]
        answers = []

        for i in range(1, len(nums)):
            prefix.append(prefix[i - 1] * nums[i - 1])
            suffix.append(suffix[i - 1] * nums[len(nums) - i])

        for i in range(len(nums)):
            answers.append(prefix[i] * suffix[len(nums) - i - 1])

        return answers

File: Leetcode/238-Product-of-Array-Except-Self/238-Product-of-Array-Except-Self.cpp
class Solution {
    public:
        vector<int> productExceptSelf(vector<int>& nums) {
            vector<int> answer(nums.size(), 1);
    
            for (int i = 1; i < nums.size(); i++) {
                answer[i] = answer[i - 1] * nums[i - 1];
            }
    
            int current = 1;
    
            for (int i = nums.size() - 1; i >= 0; i--) {
                answer[i] *= current;
                current *= nums[i];
            }
    
            return answer;
        }
    };